src/main/java/org/betonquest/betonquest/modules/logger/custom/DebugLogFormatter.java

```diff
@@ -27,11 +27,12 @@ public DebugLogFormatter() {
 
     @Override
     public String format(final LogRecord record) {
+        if (!(record instanceof BetonQuestLogRecord logRecord)) {
+            throw new UnsupportedOperationException("The DebugLogFormatter can only format BetonQuestLogRecords!");
+        }
```

Comment:
This is a Liskov substitution principle violation. It has to be either fixed or at least documented.